api/src/test/java/org/openmrs/validator/AllergyValidatorTest.java

```diff
@@ -191,4 +191,12 @@ public void validate_shouldRejectNumericReactionValue() {
 		validator.validate(allergy, errors);
 		assertTrue(errors.hasErrors());
 	}
+	@Test
+	public void validate_shouldRejectNumericAllergenValue() {
+		Allergen allergen = new Allergen(AllergenType.DRUG, null , "45" );
+		allergy.setAllergen(allergen);
+		Errors errors = new BindException(allergy,"allergy");
+		validator.validate(allergy, errors);
+		assertTrue(errors.hasErrors());
```

Comment:
This should actually validate the error message.

Reply:
@ibacher Sir, please check I made the changes.

Reply:
@ibacher could you please review my PR.